src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -917,704 +988,529 @@ static void WindowFootpathSetProvisionalPathAtPoint(const ScreenCoordsXY& screen
+            else
+            {
+                // Reset legacy path, we default to a surface (if there are any)
+                gFootpathSelection.LegacyPath = OBJECT_ENTRY_INDEX_NULL;
+            }
         }
-    }
 
-    if (surfaceIndex == OBJECT_ENTRY_INDEX_NULL)
-    {
-        if (legacyPathIndex == OBJECT_ENTRY_INDEX_NULL)
+        if (surfaceIndex == OBJECT_ENTRY_INDEX_NULL)
         {
-            // No surfaces or legacy paths available
-            return false;
+            if (legacyPathIndex == OBJECT_ENTRY_INDEX_NULL)
+            {
+                // No surfaces or legacy paths available
+                return false;
+            }
+
+            // No surfaces available, so default to legacy path
+            gFootpathSelection.LegacyPath = legacyPathIndex;
         }
 
-        // No surfaces available, so default to legacy path
-        gFootpathSelection.LegacyPath = legacyPathIndex;
+        gFootpathSelection.NormalSurface = surfaceIndex;
+        gFootpathSelection.QueueSurface = queueIndex;
+        gFootpathSelection.Railings = railingIndex;
+        return true;
     }
 
-    gFootpathSelection.NormalSurface = surfaceIndex;
-    gFootpathSelection.QueueSurface = queueIndex;
-    gFootpathSelection.Railings = railingIndex;
-    return true;
-}
-
-static PathConstructFlags FootpathCreateConstructFlags(ObjectEntryIndex& type)
-{
-    PathConstructFlags pathConstructFlags = 0;
-    if (gFootpathSelection.IsQueueSelected)
-        pathConstructFlags |= PathConstructFlag::IsQueue;
-    if (gFootpathSelection.LegacyPath != OBJECT_ENTRY_INDEX_NULL)
+    PathConstructFlags CreateConstructFlags(ObjectEntryIndex& type)
     {
-        pathConstructFlags |= PathConstructFlag::IsLegacyPathObject;
-        type = gFootpathSelection.LegacyPath;
+        PathConstructFlags pathConstructFlags = 0;
+        if (gFootpathSelection.IsQueueSelected)
+            pathConstructFlags |= PathConstructFlag::IsQueue;
+        if (gFootpathSelection.LegacyPath != OBJECT_ENTRY_INDEX_NULL)
+        {
+            pathConstructFlags |= PathConstructFlag::IsLegacyPathObject;
+            type = gFootpathSelection.LegacyPath;
+        }
+        return pathConstructFlags;
     }
-    return pathConstructFlags;
-}
+};
 
+// Keyboard Shortcut wrapper functions
 void window_footpath_keyboard_shortcut_turn_left()
 {
-    rct_window* w = window_find_by_class(WC_FOOTPATH);
-    if (w == nullptr || WidgetIsDisabled(w, WIDX_DIRECTION_NW) || WidgetIsDisabled(w, WIDX_DIRECTION_NE)
-        || WidgetIsDisabled(w, WIDX_DIRECTION_SW) || WidgetIsDisabled(w, WIDX_DIRECTION_SE) || _footpathConstructionMode != 2)
-    {
-        return;
-    }
-    int32_t currentRotation = get_current_rotation();
-    int32_t turnedRotation = _footpathConstructDirection - currentRotation + (currentRotation % 2 == 1 ? 1 : -1);
-    WindowFootpathMousedownDirection(turnedRotation);
+    auto* window = dynamic_cast<FootpathWindow*>(window_find_by_class(WC_FOOTPATH));
```

Comment:
Use ```static_cast``` when we can determine compile time if the type is castable, dynamic_cast will perform checks at runtime. This applies to all the other ones aswel.

Reply:
Ok, I will change that. thanks for the explanation.